packages/ui/react-ui-menu/src/hooks.ts

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import { useSignalEffect } from '@preact/signals-react';
+import { useCallback, useState } from 'react';
+
+import { Graph, type NodeArg, type Node } from '@dxos/app-graph';
+
+import { type MenuItem, type MenuItemGroup } from './defs';
+
+type Edge = { source: string; target: string };
+
+const edgesArrayToRecord = (edges: { source: string; target: string }[]): Record<string, string[]> => {
+  return Object.fromEntries(
+    Object.entries(
+      edges.reduce((acc: Record<string, { inbound: string[]; outbound: string[] }>, { source, target }) => {
+        if (!acc[source]) {
+          acc[source] = { inbound: [], outbound: [] };
+        }
+        if (!acc[target]) {
+          acc[target] = { inbound: [], outbound: [] };
+        }
+
+        const sourceEdges = acc[source];
+        if (!sourceEdges.outbound.includes(target)) {
+          sourceEdges.outbound.push(target);
+        }
+
+        const targetEdges = acc[target];
+        if (!targetEdges.inbound.includes(source)) {
+          targetEdges.inbound.push(source);
+        }
+
+        return acc;
+      }, {}),
+    ).map(([id, { outbound }]): [string, string[]] => [id, outbound]),
+  );
+};
+
+export const useMenuGraph = (actionCreator: () => { nodes: NodeArg<any>[]; edges: Edge[] }) => {
```

Comment:
I think I'd call this `useMenuActions` and not return the `graph`. Is there a reason you were thinking the graph needs to be exposed?

Reply:
Not really and fair enough, can make that change.